src/miner.cpp

```diff
@@ -624,6 +623,104 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
+    {
+        scriptPubKey = blocknew.vtx[1].vout[1].scriptPubKey;
+        
+        // Remove the existing single stake output.
+        blocknew.vtx[1].vout.pop_back();
+    
+        nActualStakeOutputValue = nTotalStakeOutputValue / nUTXOCount;
+
+        if (fDebug2) LogPrintf("SplitCoinStakeOutput: nUTXOCount = %f", nUTXOCount);
+        if (fDebug2) LogPrintf("SplitCoinStakeOutput: nActualStakeOutputValue = %f", CoinToDouble(nActualStakeOutputValue));
+        
+        int64_t nSumStakeUTXOValue = 0;
+        for (unsigned int i = 1; i < nUTXOCount; i++)
+        {
+            blocknew.vtx[1].vout.push_back(CTxOut(nActualStakeOutputValue, scriptPubKey));
+            LogPrintf("SplitCoinStakeOutput: create stake UTXO %i value %f", i, CoinToDouble(nActualStakeOutputValue));
+            nSumStakeUTXOValue += nActualStakeOutputValue;
+        }
+        // For the last UTXO, subtract the running sum from the desired total, which does not include the last UTXO so far,
+        // to recover anything lost from rounding. This will be a very small difference from the others.
+        // The reason we go through this trouble is that integer division rounds down, and we don't even want
+        // to have the wallet lose 1 Halford.
+        blocknew.vtx[1].vout.push_back(CTxOut((nTotalStakeOutputValue - nSumStakeUTXOValue), scriptPubKey));
+        LogPrintf("SplitCoinStakeOutput: create stake UTXO %i value %f", nUTXOCount, CoinToDouble(nTotalStakeOutputValue - nSumStakeUTXOValue));
+    }
+}
+
+
+
+unsigned int GetNumberofStakeOutputs(int64_t nValue)
+{
+    int64_t nMinUTXOPostSplitValue = 0;
+    double dEfficiency = 0.0;
+    int64_t nDesiredStakeUTXOValue = 0;
+    unsigned int nUTXOCount = 1;
+
+    // Pull Minimum Post Stake UTXO Split Value from config or command line parameter.
+    // Do not allow it to be specified below 800 GRC.
+    nMinUTXOPostSplitValue = GetArg("-minimumstakeutxovalueforsplit", 800) * COIN;
+    nMinUTXOPostSplitValue = (nMinUTXOPostSplitValue < 800 * COIN) ? 800 * COIN : nMinUTXOPostSplitValue;
+
+    LogPrintf("GetNumberofStakeOutputs: nMinUTXOPostSplitValue = %f", CoinToDouble(nMinUTXOPostSplitValue));
+
+    // Pull efficiency for UTXO staking from config, but constrain to the interval [0.75, 0.98]. Use default of 0.90.
+    dEfficiency = (double)GetArg("-stakingefficiency", 90) / 100;
+    if (dEfficiency > 0.98)
+        dEfficiency = 0.98;
+    else if (dEfficiency < 0.75)
+        dEfficiency = 0.75;
+
+    LogPrintf("GetNumberofStakeOutputs: dEfficiency = %f", dEfficiency);
+
+    // Set Desired UTXO size post stake based on efficiency and difficulty, but do not allow to go below MinUTXOPostSplitValue.
+    // Note that we use GetAverageDifficulty over a 4 hour (160 block period) rather than StakeKernelDiff, because the block
+    // to block difficulty has too much scatter.
+    nDesiredStakeUTXOValue = 9942.2056 * GetAverageDifficulty(160) * (3.0 / 2.0) * (1 / dEfficiency  - 1) * COIN;
```

Comment:
please link to explanation of the constants (file under doc/ or steemit)
move 9942.2056 constant somewhere up

Reply:
That is a good point. I can do that, since I did all of these calculations... the problem is that the final blue paper is not done yet, only the initial draft. I will put a link to the initial draft paper.